content/2024-01-25-2023-suave-tee-coprocessor.mdx

```diff
@@ -0,0 +1,406 @@
+---
+slug: suave-tee-coprocessor
+title: "Sirrah: Speedrunning a TEE Coprocessor"
+authors: [amiller]
+tags: [suave, tee]
+image: /img/posts/suave-tee-coprocessor/sirrah-title.png
+hide_table_of_contents: false
+date: 2024-01-25 00:00
+description: "Get in losers, we are attaching a TEE to a blockchain. All in under 2k lines of code, mostly Solidity"
+---
+
+This post presents Sirrah, a minimalist demo of extending a blockchain with confidential computing using Trusted Execution Environments (TEEs). We'll cover the development end-to-end: starting from Gramine-SGX and ordinary REVM, and ending with MEV-aware auction application... and a timelock encryption demo that you [can try right now](./todo).
+
+<!--truncate-->
+
+*Authors: Andrew Miller, Mateusz Morusiewicz, Frieder Paape.*
+
+![andromeda sirrah overview](../static/img/sirrah-splash.png)
+
+## What is a TEE Coprocessor, did you just make that up?
+For our purposes, a TEE coprocessor is a system that runs adjacent to an ordinary blockchain, and extends that blockchain's smart contract environment with TEE-based secure computing.
+
+
+By the end of the post, the functionality we’ll end up with is similar to other Secret Network and other TEE-based smart contracts like Oasis, Phala, Obscuro, as well as earlier research papers like [Ekiden](https://arxiv.org/pdf/1804.05141.pdf) and [State for the Stateless](https://eprint.iacr.org/2017/201). All of these share the same high level design: take an ordinary blockchain, but make the smart contract VM run inside a TEE. Now you have a smart contract system with confidentiality of some kind. There’s actually a [Systemization of Knowledge paper](https://arxiv.org/abs/2203.08548) including 20+ variations of this throughout academia and industry.
```

Comment:
Suggestion:
By the end of the post, the functionality we’ll end up with is similar to Secret Network and other TEE-based smart contracts like Oasis, Phala, Obscuro